### PR TITLE
Feature/pro 1782 simple plotting distribution skews

### PR DESCRIPTION
**Pull request by boba-and-beer**

### CODE REVIEW

File: relevanceai/operations/cluster/cluster.py [hunk 1]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+            # Get the average in the score too
+            mean = sample_comm_df[numeric_field].mean()
+            std = sample_comm_df[numeric_field].var()
+            plt.title(
+                community + str(f" - average: {round(mean, 2)}, var: {round(std, 2)}")
+            )
+            plt.show()
+
+    @track
+    def plot_distributions_by_measure(
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:675]: same here

File: relevanceai/operations/cluster/cluster.py [hunk 2]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+            if i == top_indices:
+                return
+            sample_comm_df = df[df[cluster_field] == community]
+            g = sns.displot(
+                sample_comm_df[numeric_field],
+            )
+            g.set(xlim=(facet_result["min"], facet_result["max"]))
+            plt.title(community + str(f" - {measurement_name}: {measurement}"))
+
+    def plot_most_skewed(
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:741]: same here

File: relevanceai/operations/cluster/cluster.py [hunk 3]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+            measure_function=skew,
+            top_indices=top_indices,
+            dataset_id=dataset_id,
+            asc=asc,
+        )
+
+    def _check_for_dataset_id(self):
+        if not hasattr(self, "dataset_id"):
+            raise ValueError(
+                "You are missing a dataset ID. Please set using the argument dataset_id='...'."
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:764]: Message should in `relevanceai.constants.messages`

File: relevanceai/operations/cluster/cluster.py [hunk 4]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+            )
+
+    def _get_cluster_field_name(self, alias: str = None):
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:767]: This is good

File: relevanceai/operations/cluster/cluster.py [hunk 5]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+                fields=[cluster_field],
+                page_size=int(self.config["data.max_clusters"]),
+                page=1,
+                asc=True,
+            )
+            if "results" in facet_results:
+                facet_results = facet_results["results"]
+            if cluster_field not in facet_results:
+                raise ValueError(
+                    f"No clusters with alias `{alias}`. Please check the schema."
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:823]: put in messages

File: relevanceai/operations/cluster/cluster.py [hunk 6]
```diff
@@ -628,3 +637,196 @@ def aggregate(
...
+        if isinstance(self.vector_fields, list):
+            set_cluster_field = f"_cluster_.{'.'.join(self.vector_fields)}.{alias}"
+        elif isinstance(self.vector_fields, str):
+            set_cluster_field = f"{self.cluster_field}.{self.vector_fields}.{alias}"
+        return set_cluster_field
+
+    def list_cluster_ids(
+        self,
+        alias: str = None,
+        minimum_cluster_size: int = 3,
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:779]: I dont understand why there is minimum?

**Review by jtwinrelevanceai — CHANGES_REQUESTED** [review]

File: relevanceai/operations/cluster/cluster.py [hunk 7]
```diff
@@ -628,3 +637,196 @@ def aggregate(
             asc=asc,
             flatten=flatten,
         )
+
+    @track
+    def plot_distributions(
```
[the remark this answers is not in the document]
Comment (boba-and-beer) [reply relevanceai/operations/cluster/cluster.py:642]: Moved to `visualise/cluster.py`

File: relevanceai/operations/cluster/cluster.py (hunk 2 above)
Comment (boba-and-beer) [reply relevanceai/operations/cluster/cluster.py:741]: Moved to `visualise/cluster.py`

File: relevanceai/operations/cluster/cluster.py (hunk 3 above)
Comment (boba-and-beer) [reply relevanceai/operations/cluster/cluster.py]: Moved to `relevanceai.constants.errors`

File: relevanceai/operations/cluster/cluster.py (hunk 6 above)
Comment (boba-and-beer) [reply relevanceai/operations/cluster/cluster.py]: This is when you don't care about plotting clusters when they are smaller than 3.

File: relevanceai/constants/errors.py [hunk 8]
```diff
@@ -103,3 +102,28 @@ class RegionNotFoundError(RelevanceAIError):
...
+
+class MissingClusterError(RelevanceAIError):
+    """Error for missing clusters"""
+
+    def __init__(self, alias, *args, **kwargs):
+        message = f"No clusters with alias `{alias}`. Please check the schema."
+        super().__init__(message)
+
+
+class MissingPackageError(RelevanceAIError):
```
Comment (jtwinrelevanceai) [inline relevanceai/constants/errors.py:121]: `MissingSeabornError` or

File: relevanceai/constants/errors.py [hunk 9]
```diff
@@ -103,3 +102,28 @@ class RegionNotFoundError(RelevanceAIError):
...
+    def __init__(self, package, version: Optional[str] = None, *args, **kwargs):
+        if version is not None:
+            message = f"You need to install seaborn! `pip install {package}`."
```
Comment (jtwinrelevanceai) [inline relevanceai/constants/errors.py:124]: or `message = f"You need to install {package}! pip install {package}."`

File: relevanceai/operations/cluster/cluster.py [hunk 10]
```diff
@@ -119,6 +124,9 @@ def _get_package(self, model):
         elif "community_detection" in model_name:
             package = "sentence-transformers"
 
+        else:
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/cluster/cluster.py:127]: This logic was already handled 

```
 model_name = str(model.__class__)
        if "function" in model_name:
            model_name = str(model.__name__)
```

File: relevanceai/operations/visualise/cluster.py [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+        vector_fields: Optional[List[str]] = None,
+        alias: Optional[str] = None,
+        dataset_id: Optional[str] = None,
+        **kwargs,
+    ):
+        self.vector_fields = vector_fields
+        self.alias = alias  # type: ignore
+        self.dataset_id = dataset_id
+        super().__init__(
+            credentials,
```
Comment (jtwinrelevanceai) [inline relevanceai/operations/visualise/cluster.py:34]: `credentials=credentials`

**Review by jtwinrelevanceai — CHANGES_REQUESTED** [review]
lgtm

**Review by jtwinrelevanceai — APPROVED** [review]
lgtm

**Pull request merged**